### ISSUE TITLE
[Feature Request][Android/iOS]Better cheat handling for PPSSPP

### ISSUE THREAD

**Issue by HiImBrala** [hrydgard/ppsspp#12808]
Since eternity of PPSSPP,cheats were always had to be managed through 3rd party apps (i.e ES File Explorer)
Maybe if we get in-app cheat editor?

**Comment by Emulatorer** [comment]
Related #6356

**Comment by LunaMoo** [comment]
Not really. That old issue is about cheat menu overwriting files edited elsewhere before exiting cheat menu, which was worked around by exiting cheat menu while opening external editor. It's still opened just because a proper solution would be to detect changes in the ini files before closing cheat menu, but it's not an easy thing to do and nobody bothers with the workaround being fine as it is.

 This issue is about completely different thing, basically implementing text editor inside PPSSPP which no platform has, and as of now nobody cared to do anything about it, precisely because all platforms has their own text editors to begin with and they are better and more advanced than some basic barebones text editor in PPSSPP could ever be.

 It is more bothersome to edit cheats on mobiles since there we don't have an option to open ini file in external text editor and this could be easier to do than writing text editor for PPSSPP, but mobile platforms don't have any standard / system text editors if I recall the reason why it's not done there.

**Comment by i30817** [comment]
Speaking of cheats, i'd like if the libretro interface for them was filled in. The actual contents of the files can be transliterated later into libretro-database, but currently ppsspp in libretro frontends is a bit of a odd duck in being one of the cores where you have to edit external files.

I'm sure this was considered before because the core even has a option to 'internal cheats support' so I kind of feel i'm being presumptuous to request it on a issue, but this one made me think of it.

**Comment by LunaMoo** [comment]
PPSSPP is not a libretro project. If something doesn't work with it, it's for libretro maintainers/contributors to fix it and do it in a way that does not affect proper PPSSPP in any way.

**Comment by unknownbrackets** [comment]
It would not really be hard to have an "Add Cheat" feature that allowed you to type in a name and a few lines to add to the cheat file.  That said, I feel like that experience will not be great - just like LunaMoo says.

In theory, another app or webapp could use PPSSPP's API to help users create cheats (I probably should add more APIs to make this easier still.)  Even that, I think would be better housed in a separate app probably - it's nice for such tools to have a separate update lifecycle, and the people really excited about maintaining cheats might not be the same people who are super good at C++.

Forcing the cheat editor to be part of PPSSPP just limits the people who can contribute to it.

That said, storage might be a separate problem.  Possibly solved by having an API to update the cheat file.  Then even a webapp could be the cheat editor.  Lots more people know how to write webapps than C++, so that'd probably result in a much more awesome cheat creator for mobile devices.

Or sure, we could have a kinda crappy one built into PPSSPP that no one really maintains often.  And isn't very cool because the people who could modify it are more interested in making PPSSPP emulate games better or faster or more accurately.  If you hate awesome things, that's an option too.

-[Unknown]

**Comment by HiImBrala** [comment]
> It would not really be hard to have an "Add Cheat" feature that allowed you to type in a name and a few lines to add to the cheat file. That said, I feel like that experience will not be great - just like LunaMoo says.
> 
> In theory, another app or webapp could use PPSSPP's API to help users create cheats (I probably should add more APIs to make this easier still.) Even that, I think would be better housed in a separate app probably - it's nice for such tools to have a separate update lifecycle, and the people really excited about maintaining cheats might not be the same people who are super good at C++.
> 
> Forcing the cheat editor to be part of PPSSPP just limits the people who can contribute to it.
> 
> That said, storage might be a separate problem. Possibly solved by having an API to update the cheat file. Then even a webapp could be the cheat editor. Lots more people know how to write webapps than C++, so that'd probably result in a much more awesome cheat creator for mobile devices.
> 
> Or sure, we could have a kinda crappy one built into PPSSPP that no one really maintains often. And isn't very cool because the people who could modify it are more interested in making PPSSPP emulate games better or faster or more accurately. If you hate awesome things, that's an option too.
> 
> -[Unknown]

So will i close this or what,im more interested in seeing more games than editor for cheats since i use them for patches only.
